CONTRIBUTING.md

```diff
@@ -145,32 +261,490 @@ for review, which allows you to sidestep this issue.
+    },
+    'gitGraph': {
+        'showCommitLabel':true,
+        'mainBranchName': 'master',
+        'rotateCommitLabel': true
+    }
+} }%%
+gitGraph
+    commit id:" "
+    branch staging-next
+    branch staging
+
+    checkout master
+    checkout staging
+    checkout master
+    commit id:"    "
+    checkout staging-next
+    merge master id:"automatic"
+    checkout staging
+    merge staging-next id:"automatic "
+
+    checkout staging-next
+    merge staging type:HIGHLIGHT id:"manual"
+    commit id:"fixup"
+
+    checkout master
+    checkout staging
+    checkout master
+    commit id:"       "
+    checkout staging-next
+    merge master id:"automatic  "
+    checkout staging
+    merge staging-next id:"automatic   "
+
+    checkout staging-next
+    commit id:"fixup "
+    checkout master
+    merge staging-next type:HIGHLIGHT id:"manual (PR)"
+```
+
+
+Here's an overview of the different branches:
+
+| branch | `master` | `staging` | `staging-next` |
+| --- | --- | --- | --- |
+| Used for development | :heavy_check_mark: | :heavy_check_mark: | :x: |
+| Built by Hydra | :heavy_check_mark: | :x: | :heavy_check_mark: |
+| [Mass rebuilds][mass-rebuild] | :x: | :heavy_check_mark: | :warning: Only to fix Hydra builds |
+| Critical security fixes | :heavy_check_mark: for non-mass-rebuilds | :x: | :heavy_check_mark: for mass-rebuilds |
+| Automatically merged into | `staging-next` | - | `staging` |
+| Manually merged into | - | `staging-next` | `master` |
+
+The staging workflow is used for all main branches, `master` and `release-YY.MM`, with corresponding names:
+- `master`/`release-YY.MM`
+- `staging`/`staging-YY.MM`
+- `staging-next`/`staging-next-YY.MM`
+
+# Conventions
+
+## Branch conventions
+<!-- This section is relevant to both contributors and reviewers -->
+[branch]: #branch-conventions
+
+Most changes should go to the `master` branch, but sometimes other branches should be used instead.
+Use the following decision process to figure out which one it should be:
+
+Is the change [acceptable for releases][release-acceptable] and do you wish to have the change in the release?
+- No: Use the `master` branch, do not backport the pull request.
+- Yes: Can the change be implemented the same way on the `master` and release branches?
+  For example, a packages major version might differ between the `master` and release branches, such that separate security patches are required.
+  - Yes: Use the `master` branch and [backport the pull request](#backporting-changes).
+  - No: Create separate pull requests to the `master` and `release-XX.YY` branches.
+
+Furthermore, if the change causes a [mass rebuild][mass-rebuild], use the appropriate staging branch instead:
+- Mass rebuilds to `master` should go to `staging` instead.
+- Mass rebuilds to `release-XX.YY` should go to `staging-XX.YY` instead.
+
+See [this section][staging] for more details about such changes propagate between the branches.
+
+### Changes acceptable for releases
```

Comment:
This section was titled differently before, but I also rewrote its contents to make more sense